torch/profiler/profiler.py

```diff
@@ -44,6 +47,19 @@ def supported_activities():
     return torch.autograd._supported_activities()
 
 
+def _optimizer_post_hook(optimizer, args, kwargs):
+    KinetoStepTracker.increment_step("Optimizer")
+
+
+def profile_using_dynolog():
+    """
+    To enable tracing via dynolog we register a global optimizer step post
+    hook. Requires the 'USE_KINETO_DAEMON' environment variable to be set.
+    """
+    if os.environ.get("KINETO_USE_DAEMON", None) is not None:
```

Comment:
Is the user expected to call this function? If so, I would expect that an error is raised if it doesn't do anything.
If this should always be called, then you can just make this run as part of the profiler.py import.

Reply:
the user is not supposed to call the profile using dynolog function. Changed the syntax to `_profile_using_dynolog`.